app/jobs/delete_job.rb

```diff
@@ -0,0 +1,9 @@
+class DeleteWorker < ActiveJob::Base
+  queue_as :default
+
+  include Sidekiq::Worker
+
+  def perform(catalogue_delete)
+    catalogue_delete.each{|c| Catalogue.find_by(catalogue_num:c).destroy}

```

Comment:
Consider renaming `catalogue_delete` to `catalogue_nums` and change line 7 to the more succinct:

```ruby
Catalogue.where("catalogue_num in ?", catalogue_nums).destroy_all
```